### ISSUE TITLE
ThemeProvider: Deprecate redundant day and night color modes

### ISSUE THREAD

**Issue by siddharthkp** [primer/react#3246]
### Description

primer/react supports more options for color modes than primer/primitives does

color modes for primer/primitives: `auto | light | dark`
color modes for primer/react: `auto | day | night | light | dark`

this is because `day` and `night` color modes are equivalent to `light` and `dark` which causes confusion. We should deprecate them in the next major.

[Usage in dotcom →](https://gh.io/primer-query?name=themeprovider&attribute=colormode)

### Version

v35

**Comment by colebemis** [comment]
I wonder if we should hold off from making any breaking changes to ThemeProvider until we have a new theming API implemented with the new styling architecture. Seems unnecessary to make consumers update their ThemeProvider in v36 if it's going to change shortly after.

**Comment by joshblack** [comment]
Hey you two! Just wanted to double-check with you all if this would be a good one to include or defer for v36 👀 cc @siddharthkp @colebemis

**Comment by siddharthkp** [comment]
Still relevant :)

**Comment by siddharthkp** [comment]
Still relevant 😭

cc @francinelucca, we just paired on this, good candidate to absorb?

**Comment by francinelucca** [comment]
Looks like this is [on deck](https://github.com/github/primer/issues/5430#issuecomment-3662877144) for v39! added as a subissue as well ❤️

**Comment by mattcosta7** [comment]
While we're here - if we're making signficiant changes to ThemeProvider (especially outside the `styled-react` paths

`legacy-theme` and `colorSchemes` are still used in the main ThemeProvider via the `colorSchemes` and theme objects. These are deepmerged at startup, which evaluates and parses a 270K object, then merges that with another object, for codepaths that _should_ be entirely legacy without sx and themeget.



See:

https://github.com/primer/react/tree/main/packages/react/src/legacy-theme/ts/colors.ts

https://github.com/primer/react/blob/2a67c191a0038df64d2225c0bc11c020c2def03c/packages/react/src/ThemeProvider.tsx#L51-L54

https://github.com/primer/react/blob/2a67c191a0038df64d2225c0bc11c020c2def03c/packages/react/src/ThemeProvider.tsx#L139-L166

https://github.com/primer/react/blob/2a67c191a0038df64d2225c0bc11c020c2def03c/packages/react/src/theme.ts#L72

We can probably remove these entirely, exposing only the colorsScheme name and the mode- removing ~40-50% of our shipped bundle and more importantly improving startup costs dramatically

A demo in github/github-ui of an incomplete removal shows pretty massive wins on bundle output - 
https://github.com/github/github-ui/pull/27207#issuecomment-5012765029

**Comment by siddharthkp** [comment]
That would be great! we just need to make sure there are no instances relying on that merged theme object

**Comment by mattcosta7** [comment]
> That would be great! we just need to make sure there are no instances relying on that merged theme object

Yes @siddharthkp , definitely.  AFAICT with a few quick searches, nothing in the main bundles seems to use it directly.  in github-ui, the only usage is to enumerate the keys for colorScheme names in a dev tool, which can easily become a static list instead - we rarely change them

**Comment by lesliecdubs** [comment]
@siddharthkp is there a good argument to add this to scope or make a stretch goal for https://github.com/github/primer/issues/6761? Goals seem similar/aligned, and landing the [proposed perf wins](https://github.com/primer/react/issues/3246#issuecomment-5012781543) would be a nice bonus.

**Comment by siddharthkp** [comment]
I don't think it's a natural fit, but we can still include it we want to get it done

**Comment by lesliecdubs** [comment]
Fair. The other option is maybe @jonrohan wants to pick this up as a meaningful piece of his perf work?
